parachain/pallets/outbound-queue/src/lib.rs

```diff
@@ -308,10 +310,19 @@ pub mod pallet {
 			let queued_message: QueuedMessage =
 				versioned_queued_message.try_into().map_err(|_| Unsupported)?;
 
-			let pricing_params = T::PricingParameters::get();
-
-			let next_nonce = Nonce::<T>::get(queued_message.channel_id).saturating_add(1);
+			// Obtain next nonce
+			let nonce = <Nonce<T>>::try_mutate(
+				queued_message.channel_id,
+				|nonce| -> Result<u64, ProcessMessageError> {
+					if *nonce == u64::MAX {
+						return Err(Unsupported)
+					}
+					*nonce = nonce.saturating_add(1);
```

Comment:
Could be simplified with something like `checked_add(1).ok_or(Error::<T>::Overflow)?`